proposal/default-on-private-variables.md

```diff
@@ -15,36 +18,67 @@
 
 > This section is non-normative.
 
-By using [`!default`], a module's variable value becomes overridable by any file
-that loads the module. The primary use case is to allow module authors to
-provide default values for variables that can be overridden by users when
-importing the module.
+By using [`!default`], module authors can define a module variable with a
+default value. This variable can be overridden by any file that loads the module
+by setting a new value in the [`with`] clause. Additionally, `!default` allows
+conditionally assigning a value to a variable only when it has a `null` value,
+similar to the null-coalescing operator in other programming languages.
 
 [`!default`]: ../accepted/module-system.md#configuring-libraries
+[`with`]: ../accepted/module-system.md#configuring-libraries
 
-However, the current behavior unintentionally allows !default to affect private
-variables. This is inconsistent with the expected privacy of Sass modules.
+However, the current behavior unintentionally allows the `with` clause to modify
+the value of a private variable. This is inconsistent with the expected privacy
+of Sass modules.
 
 ## Summary
 
 > This section is non-normative.
 
 This proposal seeks to rectify the unintended behavior of the `!default` flag
-affecting private variables in Sass modules. By disallowing `!default` on
-private variables, we reinforce module encapsulation and prevent unexpected
-overrides of internal module states.
+affecting private variables in Sass modules. By disallowing private variables
+from being overridden in a `with` clause, we reinforce module encapsulation and
+prevent unexpected overrides of internal module states.
 
 ## Semantics
 
-Add the following to [the Semantics for "Executing a Variable Declaration"]
-after validating that `resolved` is not a variable from a built-in module:
+### `@use`
 
-[the Semantics for "Executing a Variable Declaration"]: ../spec/variables.md#executing-a-variable-declaration
+Replace way each [`KeywordArgument`] in a `WithClause` is processed with:
 
-* Otherwise, if `name` begins with `-` and `declaration` has a
-  `!default` flag, throw an error.
+* For each `KeywordArgument` `argument` in this clause:
 
-  > Sass treats `_` and `-` as equivalent in variable names.
+  * If the `argument`'s identifier name begins with `-`, throw an error.
+
+    > Sass treats `_` and `-` as equivalent in identifiers.
+
+  * Otherwise, let `value` be the result of evaluating `argument`'s expression.
+
+  * Add a variable to `rule-config` with the same name as `argument`'s
+    identifier and with `value` as its value.
```

Comment:
You don't necessarily need to repeat this here; you could just say "Add the following line before evaluating a `KeywordArgument`'s expression:" and then add the error-throwing clause.